src/components/Chart/convertSeries.ts

```diff
@@ -20,20 +21,59 @@ export function convertSeries<XDomain>(
     })
   })
 
+  const addTick = (x: XDomain): XDomain => {
+    if (typeof x === 'number') return ((x + 1) as unknown) as XDomain
+    if (x instanceof Date) return (new Date(x.getTime() + 1) as unknown) as XDomain
+    if (typeof x === 'object' && x !== null) return ((+x + 1) as unknown) as XDomain
+    return x
+  }
+
+  const hiddenRanges = rangeAreas?.filter((r) => r.mask?.show && !r.mask?.showDots) ?? []
+
+  const isHidden = (x: XDomain): boolean => {
+    return hiddenRanges.some((r) => {
+      if (typeof x === 'string') {
```

Comment:
Make this check outside the loop. This is a O(n x m) loop, take care with what you're doing inside it.